Sources/TuistKit/Services/Cache/CacheWarmService.swift

```diff
@@ -41,7 +43,50 @@ final class CacheWarmService {
         }
     }
 
+    private func cacheFlavor(named flavorName: String?, from config: Config) throws -> TuistCore.Cache.Flavor {
+        let flavors = config.cache.flavors
+        switch flavorName {
+        case .none:
+            guard let defaultFlavor = flavors.first else {
+                throw CacheWarmServiceError.missingDefaultFlavor
+            }
+            return defaultFlavor
+
+        case let .some(name):
+            guard let flavor = flavors.first(where: { $0.name == name }) else {
+                throw CacheWarmServiceError.missingFlavor(name: name, availableFlavors: flavors.map(\.name))
+            }
+            return flavor
+        }
+    }
+
     private var currentPath: AbsolutePath {
         FileHandler.shared.currentPath
     }
 }
+
+enum CacheWarmServiceError: FatalError, Equatable {
+    case missingDefaultFlavor
+    case missingFlavor(name: String, availableFlavors: [String])
+
+    /// Error description.
+    var description: String {
+        switch self {
+        case let .missingFlavor(name, availableFlavors):
+            return "The flavor '\(name)' is missing in your project's configuration. Available cache flavors: \(availableFlavors.joined(separator: ", "))."
```

Comment:
I believe [Swift](https://developer.apple.com/documentation/foundation/listformatter) has a cool formatter for formatting lists :)